azurelinuxagent/ga/agent_update_handler.py

```diff
@@ -151,211 +127,55 @@ def __get_agent_family_manifests(self, goal_state):
-        return True
-
-    @staticmethod
-    def __log_event(level, msg, success=True):
-        if level == LogLevel.INFO:
-            logger.info(msg)
-        elif level == LogLevel.WARNING:
-            logger.warn(msg)
-        elif level == LogLevel.ERROR:
-            logger.error(msg)
-        add_event(op=WALAEventOperation.AgentUpgrade, is_success=success, message=msg, log_event=False)
-
     def run(self, goal_state):
+
         try:
             # Ignore new agents if update is disabled. The latter flag only used in e2e tests.
-            if not self._autoupdate_enabled or not conf.get_download_new_agents():
-                return
-
-            self._gs_id = goal_state.extensions_goal_state.id
-            agent_family = self.__get_agent_family_manifests(goal_state)
-            requested_version = self.__get_requested_version(agent_family)
-            agent_manifest = None  # This is to make sure fetch agent manifest once per update
-            warn_msg = ""
-            if requested_version is None:
-                # Do not proceed with update if self-update needs to download the manifest again with in an hour
-                if not self.__should_agent_attempt_manifest_download():
-                    return
-                if conf.get_enable_ga_versioning():  # log the warning only when ga versioning is enabled
-                    # TODO: Need to revisit this msg when version is missing in Goal state. We may need to handle better way to report the error
-                    warn_msg = "Missing requested version in agent family: {0} for incarnation: {1}, fallback to largest version update".format(self._ga_family, self._gs_id)
-                    GAUpdateReportState.report_error_msg = warn_msg
-                agent_manifest = goal_state.fetch_agent_manifest(agent_family.name, agent_family.uris)
-                requested_version = self.__get_largest_version(agent_manifest)
-                self._is_requested_version_update = False
-            else:
-                self._is_requested_version_update = True
-                # Save the requested version to report back
-                GAUpdateReportState.report_expected_version = requested_version
-                # Remove the missing requested version warning once requested version becomes available
-                if "Missing requested version" in GAUpdateReportState.report_error_msg:
-                    GAUpdateReportState.report_error_msg = ""
-
-            # Check if an update is allowed and update next upgrade times even if no new agent is available for upgrade
-            if not self.__check_if_agent_update_allowed_and_update_next_upgrade_times(requested_version):
+            if not conf.get_autoupdate_enabled() or not conf.get_download_new_agents():
                 return
 
-            if requested_version == CURRENT_VERSION:
+            # verify if agent update is allowed this time (RSM checks 1 hr interval; self-update checks manifest download interval)
```

Comment:
RSM updates need to be checked on every goal state, the 1-hour timer is just for retries

Reply:
I had concern we do the update on extension goal states too If we go with every goal state

Reply:
Not sure I understand your concern. We can go over it offline if you want.

My understanding is that the contract with CRP is that the Agent must process the goal state requesting the update, not 1 hour later (or less, depending on timing)

Reply:
sure, we can discuss in sync. 

My point, how do you GS related to Agent update request not the extension operation request. If we allow on every goal state, for example if we get 10 new GS within 1-hour timer, we end up fetching the agent family 10 times and validate that version is already updated or not which is unnecessary. It seems there is some tradeoffs we need to do. If we want this thing, we have to pay the price of doing extra checks